plugins/mmanon/mmanon.c

```diff
@@ -388,73 +389,83 @@ getHexVal(char c)
 }
 
 
-/* returns -1 if no integer found, else integer */
-static int64_t
-getPosInt(const uchar *const __restrict__ buf,
+/* returns 1 if valid IPv4 digit, 0 if not */
+static int
+isPosByte(const uchar *const __restrict__ buf,
 	const size_t buflen,
 	size_t *const __restrict__ nprocessed)
 {
-	int64_t val = 0;
+	int val = 0; /* Default means no byte found */
 	size_t i;
-	for(i = 0 ; i < buflen ; i++) {
-		if('0' <= buf[i] && buf[i] <= '9')
-			val = val*10 + buf[i]-'0';
-		else
+	for(i = 0 ; i < buflen; i++) {
+		if('0' <= buf[i] && buf[i] <= '9') {
+			/* Maximum 3 digits for single IPv4 Number, we only copy up to 4 numbers
+			 * but process forward to non digits */
+			if (i < 4) {
+				val = val*10 + buf[i]-'0';
+			}
+		} else
 			break;
 	}
 	*nprocessed = i;
-	if(i == 0)
-		val = -1;
-	return val;
+	/* Return 1 if more than 1 and less the 4 digits and between 0 and 255 */
+	if(	i > 0 &&
+		i < 4 &&
+		(val >= 0 && val <= 255)) {
+		return 1;
+	} else {
+		return 0;
+	}
 }
 
 /* 1 - is IPv4, 0 not */
-
 static int
 syntax_ipv4(const uchar *const __restrict__ buf,
 	const size_t buflen,
 	size_t *const __restrict__ nprocessed)
 {
-	int64_t val;
-	size_t nproc;
+	size_t nproc = 0;
 	size_t i;
 	int r = 0;
-
-	val = getPosInt(buf, buflen, &i);
-	if(val < 0 || val > 255)
+	if(	isPosByte(buf, buflen, &i) == 0) {
 		goto done;
-
-	if(i >= buflen || buf[i] != '.') {
+	}
+	if(	i >= buflen || buf[i] != '.') {
 		goto done;
 	}
 	i++;
-	val = getPosInt(buf+i, buflen-i, &nproc);
-	if(val < 0 || val > 255)
+	if(	isdigit(buf[i]) == 0 ||
+		isPosByte(buf+i, buflen-i, &nproc) == 0) {
 		goto done;
+	}
 	i += nproc;
 
 	if(i >= buflen || buf[i] != '.') {
 		goto done;
 	}
 	i++;
-	val = getPosInt(buf+i, buflen-i, &nproc);
-	if(val < 0 || val > 255)
+	if(	isdigit(buf[i]) == 0 ||
+		isPosByte(buf+i, buflen-i, &nproc) == 0) {
 		goto done;
+	}
 	i += nproc;
 
 	if(i >= buflen || buf[i] != '.') {
 		goto done;
 	}
 	i++;
-	val = getPosInt(buf+i, buflen-i, &nproc);
-	if(val < 0 || val > 255)
+	if(	isdigit(buf[i]) == 0 ||
+		isPosByte(buf+i, buflen-i, &nproc) == 0) {
 		goto done;
+	}
 	i += nproc;
 
 	*nprocessed = i;
 	r = 1;
-
 done:
+/*
+	DBGPRINTF("syntax_ipv4: %s Is IPV4=%s\n", buf, (r == 1 ? "YES" : "NO"));
+*/
```

Comment:
Did you mean to leave this code here commented out?

Reply:
> 

yes it is quiet a very busy output which was helpful for debugging only. 
I have removed it along with the extra spaces